src/app_service/service/contacts/service.nim

```diff
@@ -60,129 +71,99 @@ QtObject:
-    # if contact.identityImage != nil:
-      # thumbnail = contact.identityImage.thumbnail
-      # largeImage = contact.identityImage.large    
-
-    # status_contacts.saveContact(contact.id, contact.ensVerified, contact.ensName, contact.alias, contact.identicon, thumbnail, largeImage, contact.added, contact.blocked, contact.hasAddedUs, contact.localNickname)
-    status_contacts.saveContact(contact.id, contact.ensVerified, "", contact.alias, contact.identicon, thumbnail, largeImage, contact.added, contact.blocked, contact.hasAddedUs, "")
+  proc saveContact(self: Service, contact: ContactsDto) =
+    status_contacts.saveContact(contact.id, contact.ensVerified, contact.name, contact.alias, contact.identicon,
+    contact.image.thumbnail, contact.image.large, contact.added, contact.blocked, contact.hasAddedUs, contact.localNickname)
+    # we must keep local contacts updated
+    self.contacts[contact.id] = contact
 
   proc addContact*(self: Service, publicKey: string) =
-    var contact = self.getOrCreateContact(publicKey)
-    let updating = contact.added
-
-    if not updating:
+    var contact = self.getContactById(publicKey)
+    if not contact.added:
       contact.added = true
-      # discard status_chat.createProfileChat(contact.id)
     else:
       contact.blocked = false
 
     self.saveContact(contact)
-    
-    self.events.emit("contactAdded", Args())
-    # sendContactUpdate(contact.id, accountKeyUID)
-    if updating:
-      let profile = ContactsDto(
-        id: contact.id,
-        # username: contact.alias,
-        identicon: contact.identicon,
-        alias: contact.alias,
-        # ensName: contact.ensName,
-        ensVerified: contact.ensVerified,
-        # appearance: 0,
-        added: contact.added,
-        blocked: contact.blocked,
-        hasAddedUs: contact.hasAddedUs,
-        # localNickname: contact.localNickname
-      )
-      # TODO fix this to use ContactsDto
-      # self.events.emit("contactUpdate", ContactUpdateArgs(contacts: @[profile]))
+    self.events.emit(SIGNAL_CONTACT_ADDED, ContactAddedArgs(contact: contact))
 
   proc rejectContactRequest*(self: Service, publicKey: string) =
-    let contact = self.getContact(publicKey)
+    var contact = self.getContactById(publicKey)
     contact.hasAddedUs = false
 
     self.saveContact(contact)
-    self.events.emit("contactRemoved", Args())
-    # status_contacts.rejectContactRequest(publicKey)
-
-  proc changeContactNickname*(self: Service, accountKeyUID: string, publicKey: string, nicknameToSet: string) =
-    # status_contacts.setNickName(publicKey, nicknameToSet, accountKeyUID)
-    var contact = self.getOrCreateContact(publicKey)
-    # let nickname =
-    #   if (nicknameToSet == ""):
-    #     contact.localNickname
-    #   elif (nicknameToSet == DELETE_CONTACT):
-    #     ""
-    #   else:
-    #     nicknameToSet
-
-    # contact.localNickname = nickname
+    self.events.emit(SIGNAL_CONTACT_REMOVED, ContactArgs(contactId: contact.id))
```

Comment:
This will probably need an update once `base_02` is rebased on top of `master`.

Using `saveContact` is no longer correct here as `wakuexit_saveContact` has been removed from status-go. In other words, instead of calling `saveContact` for every contact related action (blocking/unblocking/adding/removing etc), there's now dedicated APIs.

So this code will break contacts functionality all over again.